packages/datastore-storage-adapter/__tests__/ExpoSQLiteDatabase.test.ts

```diff
@@ -0,0 +1,287 @@
+			jest.resetAllMocks();
+		});
+
+		it('should resolve promise and return first result', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.get(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).then(result => {
+				expect(result).toBe(MOCK_RESULT_ARRAY_1[0]);
+			});
+		});
+
+		it('should resolve promise and return undefined when no results are found', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.get(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).then(result => {
+				expect(result).toBe(undefined);
+			});
+		});
+
+		it('should reject the promise and return SQLError when executeSql encouters an error', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.get(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).catch(result => {
+				expect(result).toMatchObject(MOCK_SQL_ERROR);
+			});
+		});
+	});
+
+	describe('save() Tests', () => {
+		beforeAll(() => {
+			jest.resetAllMocks();
+		});
+
+		it('should return promise that resolves to null', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.save(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).then(result => {
+				expect(result).toBe(null);
+			});
+		});
+
+		it('should return promise that resolves to null', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.save(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).then(result => {
+				expect(result).toBe(null);
+			});
+		});
+
+		it('should reject the promise and return SQLError when executeSql encouters an error', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.save(
+				DEFAULT_STATEMENT,
+				DEFAULT_STATEMENT_PARAMS
+			).catch(result => {
+				expect(result).toMatchObject(MOCK_SQL_ERROR);
+			});
+		});
+	});
+
+	describe('batchQuery() Tests', () => {
+		beforeAll(() => {
+			jest.resetAllMocks();
+		});
+
+		const queryParameterizedStatements: Set<ParameterizedStatement> = new Set();
+		queryParameterizedStatements.add([
+			DEFAULT_STATEMENT,
+			DEFAULT_STATEMENT_PARAMS,
+		]);
+		queryParameterizedStatements.add([
+			DEFAULT_STATEMENT,
+			DEFAULT_STATEMENT_PARAMS,
+		]);
+		it('should return promise that resolves to first result element of each statement', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.batchQuery(
+				queryParameterizedStatements
+			).then(result => {
+				expect(result).toEqual([MOCK_RESULT_ARRAY_1[0], undefined]);
+			});
+		});
+
+		it('should return promise that resolves to null', () => {
+			expect.assertions(1);
+			return ExpoSQLiteAdapterInstance.batchQuery(
+				queryParameterizedStatements
+			).catch(result => {
+				expect(result).toMatchObject(MOCK_SQL_ERROR);
+			});
+		});
+	});
+
+	describe('batchSave() Tests', () => {
+		beforeAll(() => {
+			jest.resetAllMocks();
+		});
+
+		const saveParameterizedStatements: Set<ParameterizedStatement> = new Set();
+		saveParameterizedStatements.add([
```

Comment:
Since `batchSave` (and this is applicable to other functions as well) accepts multiple *or* single statements, your tests should at least cover those use cases: E.g.
* multiple save statements, no delete statements
* no save statements, multiple delete statements
* no statements at all
* multiple save and delete statements

Reply:
Thanks Chris, I will add tests for all the above cases.